src/tools/html-checker/main.rs

```diff
@@ -0,0 +1,65 @@
+use std::env;
+use std::path::Path;
+use std::process::{Command, Output};
+
+fn check_html_file(file: &Path) -> bool {
+    let mut to_mute = vec!["PROPRIETARY_ATTRIBUTE"];
+    if file.file_name().and_then(|f| f.to_str()) == Some("settings.html") {
+        // In `settings.html`, we have empty `<span>` for the toggles.
+        to_mute.push("TRIM_EMPTY_ELEMENT");
+    }
+    let to_mute_s = to_mute.join(",");
+    let mut command = Command::new("tidy");
+    command
+        .arg("-e")
+        .arg("-q")
+        .arg("--mute-id") // this option is useful in case we want to mute more warnings
+        .arg("yes")
+        .arg("--mute")
+        .arg(&to_mute_s)
+        .arg(file);
+
+    let Output { status, stderr, .. } = command.output().expect("failed to run tidy command");
+    if status.success() {
+        true
+    } else {
+        let stderr = String::from_utf8(stderr).expect("String::from_utf8 failed...");
+        if stderr.is_empty() {
+            true
```

Comment:
What is this doing? When would the command fail and have no output?

Reply:
Strangely enough, even if no warning is emitted, it returns 1. So my guess here is that the muted warnings are still counted but not displayed.

Reply:
(I'll add a comment to explain that)

Reply:
Ok - could you check specifically for status code 1 instead of ignoring all failures?

Reply:
The documentation of tidy explicitly says:

 * 0 -> no warning and no error
 * 1 -> warnings but no error
 * 2 -> errors (and maybe warnings too)
 
But you're right, I should check if it's not 2.

Reply:
> I should check if it's not 2.

No, you should check if it's not 0 or 1. If it's 2 that means there *were* errors and the build should fail.

Reply:
Yep, in this context, we return `true` if there was no issue, so it's what i said, no?

Reply:
No, what you said is different if it's a status code other than 0, 1, or 2, like if it fails to read a file.

Reply:
That's gonna be problematic then:

```
$ tidy -quiet -errors whatever
Document: "whatever" is not a file!

$ echo $?
0
```

Reply:
Ugh, that's really annoying. Do you know any other HTML linters?

Reply:
I saw a few others but they all seem much less good than `tidy`. If the file cannot be read, we can simply add a check over that too, but that will be picked by the link checker in any case. So is there really a need for that here?

Reply:
Hmm, I guess not. Checking for an exit code other than 2 seems ok for now, simulacrum may have more ideas.